src/S3/S3MultiRegionClient.php

```diff
@@ -224,17 +227,31 @@ private function determineRegionMiddleware() {
                         $region = null;
                         if (isset($result['@metadata']['headers']['x-amz-bucket-region'])) {
                             $region = $result['@metadata']['headers']['x-amz-bucket-region'];
+                            $this->cache->set($cacheKey, $region);
                         } else {
-                            /** @var S3ClientInterface $client */
-                            $client = $this->getClientFromPool();
-                            $region = (yield $client->determineBucketRegionAsync(
+                            $region = $this->getDetermineBucketRegionGenerator(
                                 $command['Bucket']
-                            ));
+                            );
                         }
 
-                        $this->cache->set($cacheKey, $region);
                         $command['@region'] = $region;
                         yield $handler($command);
+                    } catch (AwsException $e) {
+                        if ($e->getAwsErrorCode() === 'AuthorizationHeaderMalformed') {
```

Comment:
If the exception caught is not "AuthorizationHeaderMalformed", does this just make the exception "slient"?